core/build.gradle.kts

```diff
@@ -7,7 +7,11 @@ group="com.github.topjohnwu.libsu"
 android {
     namespace = "com.topjohnwu.superuser"
     defaultConfig {
-        consumerProguardFiles("proguard-rules.pro")
+        proguardFiles(
+            getDefaultProguardFile("proguard-android-optimize.txt"),
+            "proguard-rules.pro"
+        )
+        consumerProguardFiles("consumer-rules.pro")
```

Comment:
No need this config, if you want to integrate rules in `core/consumer-rules.pro`, just merge them into `core/proguard-rules.pro`, all rules will be bundled in aar.

Reply:
No, consumer rules are included into parent project, while proguard rules are applied for the library specifically. For this case, it's best to also have proguard rules in case obfuscation is enabled in the future

Reply:
I will leave this change open for an opinion of a maintainer of the library (@topjohnwu ). I don't see you in the list of contributors to take your opinion into account.

In my humble opinion adding proguard rules is not overengineering. This does not increase the complexity of the code in any way and will prevent future issues.